maker/src/orderbook_ws.rs

```diff
@@ -0,0 +1,178 @@
+                        .await
+                        .expect("to be able to connect to coordinator WS API");
+
+                // We request the filled matches for all our limit orders periodically.
+                let (task, _handle) = async move {
+                    loop {
+                        if let Err(e) = sink
+                            .send(
+                                tungstenite::Message::try_from(
+                                    OrderbookRequest::LimitOrderFilledMatches { trader_id },
+                                )
+                                .expect("valid message"),
+                            )
+                            .await
+                        {
+                            tracing::error!("Failed to ask for limit order filled matches: {e:#}");
+                        };
+
+                        tokio::time::sleep(REQUEST_FILLED_MATCHES_INTERVAL).await;
+                    }
+                }
+                .remote_handle();
+
+                tokio::spawn(task);
+
+                while let Ok(Some(msg)) = stream.try_next().await {
+                    if let Err(e) = process_message(msg, &position_manager, &trader_id).await {
+                        tracing::error!("Failed to process orderbook message: {e:#}");
+                    }
+                }
+
+                tokio::time::sleep(RECONNECT_TIMEOUT).await;
+                tracing::debug!(
+                    ?RECONNECT_TIMEOUT,
+                    "Reconnecting to orderbook WS after timeout"
+                );
+            }
+        });
+    }
+}
+
+async fn process_message(
+    msg: String,
+    position_manager: &xtra::Address<position::Manager>,
+    maker_trader_id: &PublicKey,
+) -> Result<()> {
+    tracing::trace!(%msg, "New message from orderbook");
+
+    let msg = serde_json::from_str::<OrderbookMsg>(&msg).context("Deserialization failed")?;
+
+    match msg {
+        OrderbookMsg::LimitOrderFilledMatches { trader_id, matches } => {
+            ensure!(
+                trader_id == *maker_trader_id,
+                "Got LimitOrderFilledMatches for wrong trader"
+            );
+
+            let orders = matches
+                .into_iter()
+                .map(|(order_id, contracts)| {
+                    OrderTenTenOne::new(
+                        order_id,
+                        // TODO: Get `ContractSymbol` from the orderbook.
+                        position::ContractSymbol::BtcUsd,
+                        contracts,
+                    )
+                })
+                .collect::<Vec<_>>();
+
+            tracing::info!(n = %orders.len(), "Received limit order filled matches");
+
+            let _ = position_manager.send(PositionUpdateTenTenOne(orders)).await;
+        }
+        OrderbookMsg::Match(FilledWith { order_id, .. }) => {
+            // We cannot rely directly on this message because the match does not specify the
```

Comment:
> We cannot rely directly on this message because the match does not specify the direction.

Why not? We know the `order_id` and given we created the order, we know the direction.

Reply:
I left the comment because I tried it and it didn't work. The problem stems from the fact that the `quantity` in the `Match` is absolute, so the sign is always positive and there is no way to figure out the direction.

If we remembered the orders we created as the maker we could figure it out, but that information is not persisted on the maker.

Reply:
> If we remembered the orders we created as the maker we could figure it out, but that information is not persisted on the maker.

Right ... there is this ticket for that https://github.com/get10101/10101/issues/227

Reply:
Good point!

I think it wouldn't hurt to persist the limit orders and it might be handy in this scenario, but it's still pretty limited as we only learn about the match and we cannot guarantee that the trade is executed. At least once the order is filled we know that the protocol was successfully started, so it's very likely that the trade will be executed.

The issue also mentions persisting the position, but in the current version where the coordinator is the only source of truth for this I actually thing relying solely on the orderbook/coordinator is best, as there can't be disagreements. Once we introduce maker-coordinator DLCs we can get rid of all this, and then persisting some position metadata will be more interesting.